src/app/util/attribute-storage.cpp

```diff
@@ -176,13 +176,15 @@ uint8_t emberAfGetDynamicIndexFromEndpoint(EndpointId id)
 EmberAfStatus emberAfSetDynamicEndpoint(uint8_t index, EndpointId id, EmberAfEndpointType * ep, uint16_t deviceId,
                                         uint8_t deviceVersion)
 {
-    index += FIXED_ENDPOINT_COUNT;
+    auto realIndex = index + FIXED_ENDPOINT_COUNT;
```

Comment:
auto -> int?

Reply:
I could do that, but the point is we don't really care what this gets promoted to by C's rules.  We just check that it's in-range and then cast it to the type we really want...  So I was pretty carefully avoiding using `int` there.

Reply:
It's not that we care or don't care. It's that the type is clear and it's "int". It isn't going to change and there aren't any other promotion rules like that. Why obfuscate?